webapp/channels/src/components/with_tooltip/with_tooltip_new/index.tsx

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {
+    useFloating,
+    autoUpdate,
+    offset,
+    useHover,
+    useFocus,
+    useDismiss,
+    useRole,
+    useInteractions,
+    arrow,
+    FloatingPortal,
+    autoPlacement,
+    useTransitionStyles,
+    FloatingArrow,
+} from '@floating-ui/react';
+import React, {useRef, useState, memo, useMemo} from 'react';
+import type {ElementType, ReactNode} from 'react';
+import {defineMessage} from 'react-intl';
+
+import {Constants} from 'utils/constants';
+
+import TooltipContent from './tooltip_content';
+import type {ShortcutDefinition} from './tooltip_shortcut';
+
+import './tooltip.scss';
+
+const ARROW_WIDTH = 10;
+const ARROW_HEIGHT = 6;
+const ARROW_OFFSET = 8;
+
+const TOOLTIP_REST_TIME_BEFORE_OPEN = 100;
+const TOOLTIP_APPEAR_DURATION = 250;
+const TOOLTIP_DISAPPEAR_DURATION = 200;
+
+export const ShortcutKeys = {
+    alt: defineMessage({
+        id: 'shortcuts.generic.alt',
+        defaultMessage: 'Alt',
+    }),
+    cmd: '⌘',
+    ctrl: defineMessage({
+        id: 'shortcuts.generic.ctrl',
+        defaultMessage: 'Ctrl',
+    }),
+    option: '⌥',
+    shift: defineMessage({
+        id: 'shortcuts.generic.shift',
+        defaultMessage: 'Shift',
+    }),
+};
+
+interface Props {
+    children: ReactNode;
+    title: string | ReactNode;
+    emoji?: string;
+    isEmojiLarge?: boolean;
+    hint?: string;
+    shortcut?: ShortcutDefinition;
+
+    /**
+     * The root element wrapping the trigger/children of WithTooltip.
+     * @default 'span
+     */
+    triggerAs?: ElementType;
```

Comment:
Do we finally need this now (or in a foreseeable future)? I am wary of exposing things we may not need.

Reply:
Yeah thats a good idea i will remove, lets not give more degrees for freedom early on